manager.go

```diff
@@ -541,6 +542,45 @@ type pindexLoadReq struct {
 	path, pindexName string
 }
 
+// Opens a pindex if the parent index is Hot or Warm.
+func (mgr *Manager) OpenPIndexBasedOnStatus(pindexPath string) (*PIndex, error) {
+	// change this to get only single index def
+	_, indexDefsMap, err := mgr.GetIndexDefs(false)
+	if err != nil {
+		return nil, fmt.Errorf("manager: error getting index defs: %s",
+			err.Error())
+	}
+
+	pindex := &PIndex{Path: pindexPath}
+	pindexName := PIndexNameFromPath(pindexPath)
+	indexName := IndexNameFromPIndexName(pindexName)
+	if idx, ok := indexDefsMap[indexName]; ok && idx.HibernateStatus != Cold {
+		pindex, err = OpenPIndex(mgr, pindexPath)
+		if err != nil {
+			return nil, fmt.Errorf("manager: error opening pindex: %s", err.Error())
+		}
+		return pindex, nil
+	}
+	log.Printf("manager: pindex %s is from a cold index.", pindexName)
+	// load PINDEX_META only if manager's dataDir is set
+	if mgr != nil && len(mgr.dataDir) > 0 {
```

Comment:
Since this block is getting used in many places, does it make sense to refactor this to a mgr method? or a method that takes a mgr instance.. Just check whether that makes code cleaner?